### PR TITLE
Add a notification in the page when a pop up must be opened for auth.

### PR DESCRIPTION
**Pull request by frankban**
This is an implementation of Huw's idea (see https://github.com/juju/juju-gui/compare/develop...huwshimi:popup-notification)
The style can be incrementally improved, but the notification should be visible already in all circumstances.
Fixes https://github.com/juju/juju-gui/issues/2772

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7139/

**Comment by frankban** [comment]
QA, use gijoe and gisf and try all possible login paths. In all scenarios in which logging in with USSO is involved, you should see a translucent notification at the top of the page. Logging in should work at that point by both accepting and opening the pop up window AND clicking on the link included in the notification. Here are the scenarios:
- login with userpass on gijoe;
- login with macaroons on gijoe;
- login on gisf automatically redirecting to /login;
- login on gisf deployment flow (after landing on /new);
- login on gisf with the login button (after landing on /new).

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -411,7 +411,35 @@ YUI.add('juju-gui', function(Y) {
         interactive: this.get('interactiveLogin'),
         serviceName: 'juju',
         dischargeStore: window.localStorage,
-        dischargeToken: dischargeToken
+        dischargeToken: dischargeToken,
+        visitMethod: response => {
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/app.js]: Any reason not to use a string of HTML, or react to render the notification? Perhaps in a follow up tidy up?

**Comment by Lukewh** [comment]
QA-OK Works in Safari too 👍

File: jujugui/static/gui/src/app/app.js (hunk 1 above)
Comment (frankban) [inline jujugui/static/gui/src/app/app.js]: No reason, just used the method from the original prototype from Huw. I hope that we are able to remove all this code asap, when implementing the proper login with redirection flow.

File: jujugui/static/gui/src/app/store/env/bakery.js [hunk 2]
```diff
@@ -51,20 +51,25 @@ YUI.add('juju-env-bakery', function(Y) {
...
+            non-interactive mode, in which case the visitMethod is never used,
+            as the bakery is assumed to already have the required tokens;
+          - onSuccess: an optional function to be called once the macaraq has
+            been successfully completed;
+          - setCookiePath: optional string representing the endpoint register a
+            macaroon as a cookie;
+          - cookieStore: an optional customized cookie storage;
+          - dischargeStore: an optional customized discharge storage;
+          - macaroon: an initial macaroon to be included in the storage;
+          - dischargeToken: optional token to be used when discharging.
```
Comment (jaycee) [inline jujugui/static/gui/src/app/store/env/bakery.js]: Thanks for the detailed docstring update.

File: jujugui/static/gui/src/app/store/env/bakery.js [hunk 3]
```diff
@@ -75,14 +80,8 @@ YUI.add('juju-env-bakery', function(Y) {
         } else {
           this.visitMethod = this._defaultVisitMethod;
         }
-        // If there is a cookie that may already be set, such as when coming
-        // from the storefront, use that cookie name instead of the generated
-        // one.
-        if (cfg.existingCookie) {
-          this.macaroonName = cfg.existingCookie;
```
Comment (jaycee) [inline jujugui/static/gui/src/app/store/env/bakery.js]: I'm confused why we are now throwing away the existing cooking name? Wouldn't we want to keep that?

**Comment by jaycee** [comment]
QA OK. Also have to say that is a really improved experience.

**Comment by frankban** [comment]
Thanks for the reviews and doubled QAs!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**